qa/covid19_qa/pipeline.py

```diff
@@ -16,7 +16,23 @@
 DEFAULT_SORT_MODE = "prob"
 TYPE_SORT_MODE = Literal["prob", "logit"]
 
-Instance = SquadExample
+
+@dataclass
+class Document:
+    id: str
+    text: Optional[str]  # Sometimes we don't want to hold the whole text in memory and return it via the Internet.
+    title: str
+    date: str
+    source: str
+    url: str
+
+
+class Instance(SquadExample):
+    # Sometimes we're just asking questions without testing, mostly when we evaluate.
+    # So we can set the document to null then.
+    def __init__(self, document: Optional[Document] = None, **kwargs):
+        super().__init__(title=document.title, **kwargs)
+        self.document = document
```

Comment:
In the end, subclassing worked fine. I feared some Transformers library methods but proved pointless. We keep a document here, which now has a title, URL, source, and date.